decide/src/main/java/group17/LicAnalyzer.java

```diff
@@ -203,7 +203,45 @@ public boolean lic7(InputHandler input) {
         return false;
     }
 
+    /**
+     * This function calculates Launch Interceptor Condition (LIC) number 8
+     * Utilizes helper methods in GeometryUtils.java to calculate the smallest possible radius between 3 points.
+     * The first two points are separated by A_PTS.
+     * The last two points are separated by B_PTS.
+     *
+     * @param input.NUMPOINTS,RADIUS1,X_COORD,Y_COORD,A_PTS,B_PTS
+     * @return true if three points, separated by A_PTS and B_PTS cannot be contained on or within a circle of radius RADIUS1
+     */
     public boolean lic8(InputHandler input) {
+
+        double[] X = input.X_COORD;
+        double[] Y = input.Y_COORD;
+        int A = input.A_PTS;
+        int B = input.B_PTS;
+        
+        if (input.NUMPOINTS < 2 || input.NUMPOINTS > 100) {
+            throw new IllegalArgumentException("NUMPOINTS must be between 2 and 100.");
+        }
+        if (input.RADIUS1 < 0) {
+            throw new IllegalArgumentException("RADIUS1 cannot be lower than 0.");
+        }
+        if (input.NUMPOINTS < 5) {
+            return false;
```

Comment:
Any specific reason why this check is not reached unless two exception above checks are met but not the two below?

Reply:
Not quite sure I understand, but I reasoned that if (NUMPOINTS < 2) -> Exception, and according to lic requirements (NUMPOINTS < 5) -> False.

Reply:
Resolved verbally, thus resolving conversation.